cimgen/cimgen.py

```diff
@@ -835,25 +856,22 @@ def _get_attribute_class(attribute: dict) -> str:
     return _get_rid_of_hash(name)
 
 
-def _get_attribute_type(attribute: dict, is_an_enum_class: bool) -> str:
+def _get_attribute_type(attribute: dict, is_an_enum_class: bool, primitive_classes, cim_data_type_classes) -> str:
     """Get the type of an attribute: "class", "enum", "list", or "primitive".
 
     :param attribute:        Dictionary with information about an attribute of a class.
     :param is_an_enum_class: Is this attribute an enumation?
     :return:                 Type of the attribute.
     """
```

Comment:
I've improved _get_attribute_type in https://github.com/sogno-platform/cimgen/pull/40 (see commits 3b6e9d2260909e0fc3f65a328a07faec8559bc82 and bded89c6db19288406819eedcc005ca1bfbcb4dd from 12./13.10. + 0fc5750fbca5c7b62cbdb1b9902154123dfecd29 from yesterday). But to use this in your branch more changes from my branch are necessary. Maybe it's better to wait for merging of my branch and then to rebase your branch on master.

```suggestion
def _get_attribute_type(attribute: dict, class_infos: CIMComponentDefinition) -> str:
    """Get the type of an attribute: "class", "datatype", "enum", "list", or "primitive".

    :param attribute:        Dictionary with information about an attribute of a class.
    :param class_infos:      Information about the attribute class.
    :return:                 Type of the attribute.
    """
    attribute_type = "class"
    if class_infos.is_a_datatype_class():
        attribute_type = "datatype"
    elif class_infos.is_a_primitive_class():
        attribute_type = "primitive"
    elif class_infos.is_an_enum_class():
        attribute_type = "enum"
    elif attribute.get("multiplicity") in ("M:0..n", "M:0..2", "M:1..n", "M:2..n"):
        attribute_type = "list"
    return attribute_type
```